File: LABA5/elementsPack/Person.java
package elementsPack;

import java.time.LocalDate;
import java.time.LocalDateTime;

public class Person {
    public Person(String name, LocalDateTime birthday, Color hairColor, Country nationality) {
        this.name = name;
        this.birthday = birthday;
        this.hairColor = hairColor;
        this.nationality = nationality;
    }

    private String name; //Поле не может быть null, Строка не может быть пустой
    private java.time.LocalDateTime birthday; //Поле может быть null
    private Color hairColor; //Поле может быть null
    private Country nationality; //Поле может быть null

    @Override
    public String toString() {
        return "name - " + name +
                ", birthday - " + birthday +
                ", hair color - " + hairColor +
                ", nationality - " + nationality;
    }
}

File: LABA5kt/src/elementsPack/Person.kt
package elementsPack

import java.time.LocalDateTime
import kotlin.Comparable
import kotlin.Int
import kotlin.String

/**
 * Class describing operator of the movie. Can be null
 *
 *
 * name - cannot be null or empty
 *
 *
 * birthday, hairColor, nationality - CAN be null
 */
class Person(
    private var name: String,
    private var birthday: LocalDateTime,
    private var hairColor: Color,
    private var nationality: Country
) : Comparable<Person> {
    override fun compareTo(other: Person): Int {
        return this.birthday.compareTo(other.birthday)
    }

    override fun toString(): String {
        return "name - $name" +
                ", birthday - $birthday" +
                ", hair color - $hairColor" +
                ", nationality - $nationality"
    }

    fun getBirthday(): LocalDateTime {
        return birthday
    }

    fun getHairColor(): Color {
        return hairColor
    }

    fun getNationality(): Country {
        return nationality
    }

    fun getName(): String {
        return name
    }
}